src/api/walker.ts

```diff
@@ -117,14 +117,20 @@ export class Walker<TOutput extends Output> {
         if (exclude && exclude(entry.name, path)) continue;
         this.walkDirectory(this.state, path, depth - 1, this.walk);
       } else if (entry.isSymbolicLink() && resolveSymlinks) {
-        let path = this.joinPath(entry.name, directoryPath);
+        let path = directoryPath + entry.name;
```

Comment:
Use `joinPath.joinPathWithBasePath` instead.

Note: the test starts passing by making just this change so the test is not covering everything else. Mind taking a look at that?

Reply:
`joinPathWithBasePath` isn't exported, i'll have to export it to be able to use it

Reply:
> Note: the test starts passing by making just this change so the test is not covering everything else. Mind taking a look at that?

that change on it's own fixed dir symlinks but not file symlinks, just updated the tests to also check for file symlinks